### PR TITLE
feat(provider/google): Support connection draining for LBs

### PR DESCRIPTION
**Pull request by jtk54**
Affects TCP, SSL, and L7.

### CODE REVIEW

**Comment by jtk54** [comment]
https://github.com/spinnaker/spinnaker/issues/1964

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy [hunk 1]
```diff
@@ -239,7 +239,8 @@ class UpsertGoogleHttpLoadBalancerAtomicOperation extends UpsertGoogleLoadBalanc
         Boolean differentSessionCookieTtl = existingService.getAffinityCookieTtlSec() != backendService.affinityCookieTtlSec
         Boolean differentCDN = existingService.getEnableCDN() != backendService.enableCDN
         Boolean differentPortName = existingService.getPortName() != backendService.portName
-        if (differentHealthChecks || differentSessionAffinity || differentSessionCookieTtl || differentCDN || differentPortName) {
+        Boolean differentConnectionDraining = existingService.getConnectionDraining()?.getDrainingTimeoutSec() != description.backendService?.connectionDrainingTimeoutSec
```
Comment (danielpeach) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy]: Its it just `backendService?.connectionDrainingTimeoutSec`? (Thought that HTTP load balancers had a list of backend services).

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy [hunk 2]
```diff
@@ -363,6 +363,7 @@ class GoogleHttpLoadBalancerCachingAgent extends AbstractGoogleLoadBalancerCachi
         service.affinityCookieTtlSec = backendService.affinityCookieTtlSec
         service.enableCDN = backendService.enableCDN
         service.portName = backendService.portName ?: GoogleHttpLoadBalancingPolicy.HTTP_DEFAULT_PORT_NAME
+        service.connectionDrainingTimeoutSec = backendService.connectionDraining?.drainingTimeoutSec ?: 0
```
Comment (danielpeach) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy]: The linked docs don't make it look like `0` is an allowed value here. Why not `null`? 

It won't make a difference for the caching agent, but `0` will be the populated value in the edit LB UI - and if someone tries to upsert an LB with that value, I'm guessing they'll run into a problem.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy (hunk 1 above)
Comment (jtk54) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy]: This is in a `.each` loop.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy (hunk 2 above)
Comment (jtk54) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy]: Zero is an acceptable value for the connection draining timeout.

**Pull request merged**